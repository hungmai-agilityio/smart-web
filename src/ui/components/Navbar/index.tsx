import { NavItem } from '@/ui/components/Navbar/Item';

interface NavbarProps {
  items: Array<{ name: string; url?: string }>;
  styles?: string;
}

export const Navbar = ({ items, styles }: NavbarProps) => {
  return (
    <nav className={styles}>
      {items.map((item, index) => (
        <NavItem key={index} name={item.name} url={item.url} />
      ))}
    </nav>
  );
};
