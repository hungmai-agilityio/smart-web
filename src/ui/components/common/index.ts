export * from '@/ui/components/common/Button';
export * from '@/ui/components/common/Dropdown';
export * from '@/ui/components/common/Icon';
export * from '@/ui/components/common/Photo';
export * from '@/ui/components/common/Text';
export * from '@/ui/components/common/StatsRow';
