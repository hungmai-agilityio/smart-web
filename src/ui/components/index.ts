export * from '@/ui/components/Hero';
export * from '@/ui/components/NavItem';
export * from '@/ui/components/Portfolio';
export * from '@/ui/components/Pagination';
export * from '@/ui/components/Skeleton/Card';
export * from '@/ui/components/Skeleton/Image';
export * from '@/ui/components/common';
