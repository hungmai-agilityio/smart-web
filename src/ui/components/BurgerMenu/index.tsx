'use client';

import Image from 'next/image';
import { useRef, useState } from 'react';
import { Navbar } from '@/ui/components';
import { useOutsideClick } from '@/hooks';

interface BurgerProps {
  items: Array<{ name: string; url?: string }>;
}

export const BurgerMenu = ({ items }: BurgerProps) => {
  const menuRef = useRef<HTMLDivElement>(null);
  const [menuOpen, setMenuOpen] = useState<boolean>(false);

  const toggleMenu = () => {
    setMenuOpen(!menuOpen);
  };

  useOutsideClick({
    ref: menuRef,
    handler: () => setMenuOpen(false)
  });

  return (
    <div ref={menuRef}>
      <div
        className="md:hidden flex items-center cursor-pointer"
        onClick={toggleMenu}
      >
        <Image src="/menu.svg" alt="menu" width={16} height={14} priority />
      </div>

      <Navbar
        items={items}
        styles={`md:flex gap-7 ${menuOpen ? 'block' : 'hidden'}
          md:relative absolute top-full left-0 w-full md:w-auto z-10 transition-all ease-in-out
          bg-white p-5 drop-shadow-lg md:drop-shadow-none lg:mr-28 md:mr-8`}
      />
    </div>
  );
};
