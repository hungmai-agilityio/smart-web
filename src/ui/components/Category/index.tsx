'use client';

import { memo, useCallback } from 'react';
import { useRouter, useSearchParams } from 'next/navigation';

// Constants
import { abel } from '@/constants';

// Components
import { Dropdown, ItemProps } from '@/ui/components';

interface CategoryProps {
  categories: string[];
  queryParam: string;
}

export const Category = memo(({ categories, queryParam }: CategoryProps) => {
  const router = useRouter();
  const searchParams = useSearchParams();

  const currentCategory = searchParams.get(queryParam);

  // Update category and URL when a new category is selected
  const handleCategorySelect = useCallback(
    (category: string) => {
      const params = new URLSearchParams(searchParams.toString());
      params.set(queryParam, category);
      router.push(`?${params.toString()}`, { scroll: false });
    },
    [searchParams, router, queryParam]
  );

  const dropdownItems: ItemProps[] = categories.map((category) => ({
    label: category,
    value: category
  }));

  return (
    <>
      <ul
        className={`${abel.className} my-8 md:flex hidden lg:gap-48 gap-16 justify-center`}
      >
        {categories.map((category, index) => (
          <li
            className={`uppercase cursor-pointer hover:text-second ${currentCategory === category ? 'text-second' : ''}`}
            key={index}
            onClick={handleCategorySelect.bind(null, category)}
          >
            {category}
          </li>
        ))}
      </ul>
      <div className="md:hidden flex justify-center my-8">
        <Dropdown
          items={dropdownItems}
          selectedValue={currentCategory!}
          onSelect={(value) => handleCategorySelect(value)}
        />
      </div>
    </>
  );
});
