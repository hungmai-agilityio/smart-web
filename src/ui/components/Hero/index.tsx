import { ReactNode } from 'react';

// Constants
import { TYPE } from '@/constants';

// Components
import { Button, Photo } from '@/ui/components/common';

interface HeroProps {
  imageUrl: string;
  children?: ReactNode;
  buttonName?: string;
}

export const Hero = ({ imageUrl, children, buttonName }: HeroProps) => {
  return (
    <div className="relative w-full h-hero">
      <Photo src={imageUrl} alt="Hero Background" autoSize priority />
      <div className="absolute  inset-0 flex flex-col items-center justify-center">
        {children}
        {buttonName && (
          <div className="mt-10">
            <Button variant={TYPE.PRIMARY} name={buttonName} />
          </div>
        )}
      </div>
    </div>
  );
};
