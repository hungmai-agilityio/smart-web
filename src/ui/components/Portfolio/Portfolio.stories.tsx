import { Meta, StoryObj } from '@storybook/react';

// Components
import { Portfolio } from '@/ui/components/Portfolio';
import { mockImages } from '@/mocks';

const meta: Meta<typeof Portfolio> = {
  component: Portfolio,
  tags: ['autodocs'],
  argTypes: {
    images: {
      description: 'Array of portfolio images to be displayed'
    }
  }
};

export default meta;

type Story = StoryObj<typeof Portfolio>;

export const Default: Story = {
  args: {
    images: mockImages
  }
};
