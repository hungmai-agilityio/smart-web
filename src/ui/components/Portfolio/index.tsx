import React, { Suspense } from 'react';
import { clsx } from 'clsx';

// Constants
import { abel } from '@/constants/fonts';

// Interfaces
import { IPortfolio } from '@/interface';

// Components
import { Description, Heading, Photo, SkeletonImage } from '@/ui/components';

// Utils
import { formatDate } from '@/utils/format';

interface PortfolioProps {
  images: IPortfolio[];
}

export const Portfolio = ({ images }: PortfolioProps) => {
  return (
    <div className="grid md:grid-cols-2 grid-cols-1">
      {images.map((item) => (
        <div key={item.id} className="w-full h-portfolio">
          <Photo src={item.image} alt={item.name} autoSize isBlur>
            <div className="flex w-full md:justify-end">
              <div className="lg:w-1/2">
                <Heading styles="xl:text-3xl lg:text-xl font-bold text-base">
                  {item.name}
                </Heading>
                <Description styles="lg:text-base text-sm">
                  {formatDate(new Date(item.date))}
                </Description>
                <span className="relative after:content-[''] after:absolute after:left-1/2 after:translate-y-6 after:w-10 after:h-1 after:bg-white"></span>
                <p
                  className={clsx(
                    `${abel.className} lg:text-base text-xs mt-14 max-h-24 overflow-hidden`
                  )}
                >
                  {item.detail}
                </p>

                <div className="md:flex xl:gap-10 gap-5 mt-4 hidden text-sm">
                  <div className="flex gap-3 items-center">
                    <Photo
                      src="/view-white.svg"
                      alt="view"
                      autoSize
                      styles="md:w-6 md:h-3"
                    />
                    <p>{item.view}</p>
                  </div>

                  <div className="flex gap-3 items-center">
                    <Photo
                      src="/likes-white.svg"
                      alt="likes"
                      autoSize
                      styles="md:w-icon-xs md:h-3"
                    />
                    <p>{item.favorite}</p>
                  </div>
                  <div className="flex gap-3 items-center">
                    <Photo
                      src="/cmt-white.svg"
                      alt="comment"
                      autoSize
                      styles="md:w-icon-xs md:h-3"
                    />
                    <p>{item.comment}</p>
                  </div>
                </div>
              </div>
            </div>
          </Photo>
        </div>
      ))}
    </div>
  );
};
