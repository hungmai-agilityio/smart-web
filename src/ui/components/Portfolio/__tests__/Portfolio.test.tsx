import { render, screen } from '@testing-library/react';

// Mocks
import { mockImages } from '@/mocks';

// Components
import { Portfolio } from '@/ui/components/Portfolio';

describe('Portfolio component', () => {
  it('rendering list image', () => {
    render(<Portfolio images={mockImages} />);
    const img = screen.getByAltText('Project One');
    expect(img).toBeInTheDocument();
  });

  it('should be render match to snapshot', () => {
    const { container } = render(<Portfolio images={mockImages} />);
    expect(container).toMatchSnapshot();
  });
});
