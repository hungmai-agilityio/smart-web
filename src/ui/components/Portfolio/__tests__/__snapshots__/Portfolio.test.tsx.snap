// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio component should be render match to snapshot 1`] = `
<div>
  <div
    class="grid md:grid-cols-2 grid-cols-1"
  >
    <div
      class="w-full h-portfolio"
    >
      <div
        class="relative w-full h-full"
      >
        <img
          alt="Project One"
          class="transition-opacity duration-300"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=%2Fbackground-hero.webp&w=3840&q=75"
          srcset="/_next/image?url=%2Fbackground-hero.webp&w=640&q=75 640w, /_next/image?url=%2Fbackground-hero.webp&w=750&q=75 750w, /_next/image?url=%2Fbackground-hero.webp&w=828&q=75 828w, /_next/image?url=%2Fbackground-hero.webp&w=1080&q=75 1080w, /_next/image?url=%2Fbackground-hero.webp&w=1200&q=75 1200w, /_next/image?url=%2Fbackground-hero.webp&w=1920&q=75 1920w, /_next/image?url=%2Fbackground-hero.webp&w=2048&q=75 2048w, /_next/image?url=%2Fbackground-hero.webp&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
        />
      </div>
    </div>
    <div
      class="w-full h-portfolio"
    >
      <div
        class="relative w-full h-full"
      >
        <img
          alt="Project Two"
          class="transition-opacity duration-300"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=%2Fbackground-hero.webp&w=3840&q=75"
          srcset="/_next/image?url=%2Fbackground-hero.webp&w=640&q=75 640w, /_next/image?url=%2Fbackground-hero.webp&w=750&q=75 750w, /_next/image?url=%2Fbackground-hero.webp&w=828&q=75 828w, /_next/image?url=%2Fbackground-hero.webp&w=1080&q=75 1080w, /_next/image?url=%2Fbackground-hero.webp&w=1200&q=75 1200w, /_next/image?url=%2Fbackground-hero.webp&w=1920&q=75 1920w, /_next/image?url=%2Fbackground-hero.webp&w=2048&q=75 2048w, /_next/image?url=%2Fbackground-hero.webp&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
        />
      </div>
    </div>
    <div
      class="w-full h-portfolio"
    >
      <div
        class="relative w-full h-full"
      >
        <img
          alt="Project Three"
          class="transition-opacity duration-300"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=%2Fbackground-hero.webp&w=3840&q=75"
          srcset="/_next/image?url=%2Fbackground-hero.webp&w=640&q=75 640w, /_next/image?url=%2Fbackground-hero.webp&w=750&q=75 750w, /_next/image?url=%2Fbackground-hero.webp&w=828&q=75 828w, /_next/image?url=%2Fbackground-hero.webp&w=1080&q=75 1080w, /_next/image?url=%2Fbackground-hero.webp&w=1200&q=75 1200w, /_next/image?url=%2Fbackground-hero.webp&w=1920&q=75 1920w, /_next/image?url=%2Fbackground-hero.webp&w=2048&q=75 2048w, /_next/image?url=%2Fbackground-hero.webp&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
        />
      </div>
    </div>
  </div>
</div>
`;
